src/raftstore/store/peer_storage.rs

```diff
@@ -214,16 +214,23 @@ impl EntryCache {
         }
     }
 
-    pub fn compact_to(&mut self, idx: u64) {
+    pub fn compact_to<F>(&mut self, idx: u64, mut entry_gc_fn: F)
+    where
+        F: FnMut(Entry) -> (),
+    {
         let cache_first_idx = self.first_index().unwrap_or(u64::MAX);
         if cache_first_idx > idx {
             return;
         }
         let cache_last_idx = self.cache.back().unwrap().get_index();
         // Use `cache_last_idx + 1` to make sure cache can be cleared completely
         // if neccessary.
-        self.cache
-            .drain(..(cmp::min(cache_last_idx + 1, idx) - cache_first_idx) as usize);
+        for entry in self
+            .cache
+            .drain(..(cmp::min(cache_last_idx + 1, idx) - cache_first_idx) as usize)
+        {
+            entry_gc_fn(entry);
```

Comment:
seem that the drop of entry takes too much time? can you investigate why?

I am very surprised.